### PR TITLE
Use controller-runtime for backup controller

### LINKED ISSUE #3127: Make the backup controller controller-runtime based
**User Story**

**Acceptance criteria**

* The backup controller is controller-runtime based

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3127 

**Special notes for your reviewer**:

**Documentation**:

**Release note**:

```release-note
none
```

### CODE REVIEW

File: api/pkg/controller/backup/backup_controller.go [hunk 1]
```diff
@@ -80,26 +74,14 @@ func NewMetrics() *Metrics {
 			Name:      "workers",
 			Help:      "The number of running backup controller workers",
 		}),
-		CronJobCreationTimestamp: prometheus.NewGaugeVec(prometheus.GaugeOpts{
```
Comment (alvaroaleman) [inline api/pkg/controller/backup/backup_controller.go:83]: We cant populate these anymore, as the reconciliation is now done within `reconciling.EnsureNamedObjectV2`

AFAIK we don't have any alerts that rely on these anyways

File: api/pkg/controller/backup/backup_controller.go [hunk 2]
```diff
@@ -110,243 +92,173 @@ type Controller struct {
...
-	if ownerRef := metav1.GetControllerOf(object); ownerRef != nil && ownerRef.Kind == kubermaticv1.ClusterKindName {
-		cluster, err := c.clusterLister.Get(ownerRef.Name)
-		if err != nil {
-			glog.V(4).Infof("Ignoring orphaned object '%s' from cluster '%s'", object.GetSelfLink(), ownerRef.Name)
-			return
-		}
-		c.enqueue(cluster)
-		return
+	// Cleanup cleanup jobs...
+	if err := mgr.Add(&runnableWrapper{
```
Comment (alvaroaleman) [inline api/pkg/controller/backup/backup_controller.go:171]: Not very pretty, but we can't access the `stopCh` here anymore. Has the upside thought that this doesn't get started before the caches are synced.

Alternate would be to pass the `stopCh` into the `Add`. WDYT?

Comment (mrIncompetent) [reply api/pkg/controller/backup/backup_controller.go:171]: Adding a smallish controller seems legit for this :+1:

**Pull request merged**